Add docs and integration tests for Auto-compaction snapshot status API (#10510)
apache/druid: docs/operations/api-reference.md

Returns the total size of segments awaiting compaction for the given dataSource. 
This is only valid for dataSource which has compaction enabled. 

##### GET

* `/druid/coordinator/v1/compaction/status`

> Returns the status and statistics from the latest auto compaction run of all dataSource which has/had auto compaction enabled.

Reviewer: ```suggestion
Returns the status and statistics from the latest auto compaction run of all dataSources which have/had auto compaction enabled.
```

Author: Done

> The response payload includes a list of `latestStatus` object. Each `latestStatus` represent the status for a dataSource (which has/had auto compaction enabled). 

Reviewer: ```suggestion
The response payload includes a list of `latestStatus` objects. Each `latestStatus` represents the status for a dataSource (which has/had auto compaction enabled). 
```

Author: Done

The `latestStatus` object has the following keys:
* `dataSource`: name of the datasource for this status information
* `scheduleStatus`: auto compaction scheduling status. Possible values are `NOT_ENABLED` and `RUNNING`. Returns `RUNNING ` if the dataSource has an active auto compaction config submitted otherwise, `NOT_ENABLED`

> * `bytesAwaitingCompaction`: total byte of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction)

Reviewer: ```suggestion
* `bytesAwaitingCompaction`: total bytes of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction)
```

Author: Done

> * `bytesCompacted`: total byte of this datasource that is already compacted with the spec set in the auto compaction config.

Reviewer: ```suggestion
* `bytesCompacted`: total bytes of this datasource that are already compacted with the spec set in the auto compaction config.
```

Author: Done

> * `bytesSkipped`: total byte of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.

Reviewer: ```suggestion
* `bytesSkipped`: total bytes of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.
```

Author: Done

> * `segmentCountAwaitingCompaction`: total number of segments of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction)

Reviewer: ```suggestion
* `segmentCountAwaitingCompaction`: total number of segments of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction)
```

Author: Done

> * `segmentCountCompacted`: total number of segments of this datasource that is already compacted with the spec set in the auto compaction config.

Reviewer: ```suggestion
* `segmentCountCompacted`: total number of segments of this datasource that are already compacted with the spec set in the auto compaction config.
```

Author: Done

> * `segmentCountSkipped`: total number of segments of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.

Reviewer: ```suggestion
* `segmentCountSkipped`: total number of segments of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.
```

Author: Done

> * `intervalCountAwaitingCompaction`: total number of intervals of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction)

Reviewer: ```suggestion
* `intervalCountAwaitingCompaction`: total number of intervals of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction)
```

Author: Done

> * `intervalCountCompacted`: total number of intervals of this datasource that is already compacted with the spec set in the auto compaction config.

Reviewer: ```suggestion
* `intervalCountCompacted`: total number of intervals of this datasource that are already compacted with the spec set in the auto compaction config.
```

Author: Done

> * `intervalCountSkipped`: total number of intervals of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.

Reviewer: ```suggestion
* `intervalCountSkipped`: total number of intervals of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.
```

Author: Done

##### GET

* `/druid/coordinator/v1/compaction/status?dataSource={dataSource}`

Similar to the API `/druid/coordinator/v1/compaction/status` above but filters response to only return information for the {dataSource} given. 

> Note that {dataSource} given must has/had auto compaction enabled.

Reviewer: ```suggestion
Note that {dataSource} given must have/had auto compaction enabled.
```

Author: Done
